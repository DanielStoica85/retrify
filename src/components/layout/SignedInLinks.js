import React from 'react';
import { NavLink, NavItem, Badge, Button } from 'reactstrap';
import { connect } from 'react-redux';
import { logout } from '../../actions/auth';

export const SignedInLinks = props => {
    return (
        <React.Fragment>
            <NavItem>
                <NavLink href="/retro">New Retro</NavLink>
            </NavItem>
            <NavItem>
                <NavLink href="/">
                    <Badge color="danger" pill>
                        D.S.
                    </Badge>
                </NavLink>
            </NavItem>
            <NavItem>
                <Button onClick={props.logout}>Log Out</Button>
            </NavItem>
        </React.Fragment>
    );
};

const mapDispatchToProps = dispatch => {
    return {
        logout: () => dispatch(logout())
    };
};

export default connect(
    null,
    mapDispatchToProps
)(SignedInLinks);
