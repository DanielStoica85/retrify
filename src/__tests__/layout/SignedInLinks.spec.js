import React from 'react';
import renderer from 'react-test-renderer';
import SignedInLinks from '../../components/layout/SignedInLinks';

describe('SignedInLinks component', () => {
    it('renders correctly', () => {
        const component = renderer.create(<SignedInLinks />).toJSON();
        expect(component).toMatchSnapshot();
    });
});
