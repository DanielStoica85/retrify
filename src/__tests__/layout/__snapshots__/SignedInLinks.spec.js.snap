// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SignedInLinks component renders correctly 1`] = `
Array [
  <li
    className="nav-item"
  >
    <a
      className="nav-link"
      href="/retro"
      onClick={[Function]}
    >
      New Retro
    </a>
  </li>,
  <li
    className="nav-item"
  >
    <a
      className="nav-link"
      href="/"
      onClick={[Function]}
    >
      Log Out
    </a>
  </li>,
  <li
    className="nav-item"
  >
    <a
      className="nav-link"
      href="/"
      onClick={[Function]}
    >
      <span
        className="badge badge-danger badge-pill"
      >
        D.S.
      </span>
    </a>
  </li>,
]
`;
